### DISCUSSION TITLE
Can I use an object (filter params) as key?

### DISCUSSION THREAD

**Discussion by buglessir** [TanStack/query#3138] [General]
I should use react-query for a filterable list, but can I use a big object as query key? something like:

```
{
    "pagination": {
        "current": 1,
        "pageSize": 10,
        "size": "small",
        "total": 2
    },
    "filters": {
        "pollerStatus": [
            true
        ]
    },
    "sorter": {}
}
```

then:

`const list = useQuery(['list', filters], fetcher());`

Is this best practice and logical or not? do you have any idea?

Actually, current code updates list by any changes on the filters, we store this object in local state and refetch by new changes.

**Comment by TkDodo** [comment]
Yes, this is actually best practice, because react-query will the cache each result independently. If you need it for the fetcher, put it in the key:
```
const list = useQuery(['list', filters], () => fetcher(filters))
```
